apps/validator/src/main/scala/org/lfdecentralizedtrust/splice/validator/store/ValidatorConfigProvider.scala

```diff
@@ -4,34 +4,90 @@
 package org.lfdecentralizedtrust.splice.validator.store
 
 import cats.data.OptionT
+import cats.implicits.{catsSyntaxOptionId, toBifunctorOps}
+import com.digitalasset.canton.logging.{NamedLoggerFactory, NamedLogging}
+import com.digitalasset.canton.topology.PartyId
 import com.digitalasset.canton.tracing.TraceContext
-import io.circe.Codec
-import scala.concurrent.Future
-import io.circe.generic.semiauto.*
+import io.circe.generic.semiauto.deriveCodec
+import io.circe.{Codec, Decoder}
 
-final case class ScanUrlInternalConfig(
-    svName: String,
-    url: String,
-)
+import scala.concurrent.{ExecutionContext, Future}
 
-object ScanUrlInternalConfig {
-  implicit val scanUrlCodec: Codec[ScanUrlInternalConfig] = deriveCodec[ScanUrlInternalConfig]
-}
-
-class ValidatorConfigProvider(config: ValidatorInternalStore) {
+class ValidatorConfigProvider(config: ValidatorInternalStore, val loggerFactory: NamedLoggerFactory)
```

Comment:
```suggestion
class ValidatorDynamicConfigProvider(config: ValidatorInternalStore, val loggerFactory: NamedLoggerFactory)
```

Quite off-topic to this PR, more like a general "let's consider refactoring this" comment; to a casual reader it is hard to distinguish otherwise that this is not related to the (static) config from the config files.

(cc @pasindutennage-da )